joomla/joomla-cms
# [6.3] copy api language files

Pull Request for Issue #41944 .

### Summary of Changes

Adds ability to place language files under `api/language` during installation of a component
```xml
<?xml version="1.0" encoding="UTF-8"?>
<extension type="component" method="upgrade">
	[...]
	<api>
		<files folder="api">
			<folder>src</folder>
		</files>
		<languages folder="api">
			<language tag="en-GB">language/en-GB/com_example.ini</language>
		</languages>
	</api>
</extension>

```

Use example component:
[com_mycomponent.zip](https://github.com/user-attachments/files/18469825/com_mycomponent.zip)

### Testing Instructions

Add `languages` node under API in any component then test install/uninstall
Check any other component still working

### Actual result BEFORE applying this Pull Request

language file `api\language\en-GB\com_mycomponent.ini` is not copied

### Expected result AFTER applying this Pull Request

language file `api\language\en-GB\com_mycomponent.ini` is copied and deleted during uninstall

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: There is nothing special here https://docs.joomla.org/Manifest_files
